sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -168,16 +168,12 @@ internal static (string FormattedError, string? ErrorCode, IDictionary<string, s
                 }
             }
 
-            if (response.ContentStream is MemoryStream)
+            if (response.ContentStream is MemoryStream && ContentTypeUtilities.TryGetTextEncoding(response.Headers.ContentType, out Encoding _))
```

Comment:
What does this check accomplish if we are throwing the encoding value away?

Reply:
To be sure that we are not trying to decode a non-text. Encoding itself isn't important here, cause `BinaryData` only supports UTF8.